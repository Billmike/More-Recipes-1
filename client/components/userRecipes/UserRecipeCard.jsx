import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import RecipeImage from '../common/RecipeImage';

const UserRecipeCard = (props) => {
  const { recipe, handleOpenEdit, handleOpenDelete } = props,
    { id, title, description, views, upvotes, downvotes } = recipe;

  const handleOpen = () => {
    handleOpenEdit(recipe);
  };

  return (
    <div className="col s12 m6 l3">
      <div id="recipes" className="card large">
        <Link to={`/recipes/${id}`}>
          <span className="card-title">{title}</span>
        </Link>
        <div className="card-image">
          <Link to={`/recipes/${id}`}>
            <RecipeImage />
          </Link>
        </div>
        <div className="card-content">
          <p>{description}</p>
        </div>
        <div className="card-action">
          <ul className="center-align">
            <li id="views"><i className="material-icons tiny">visibility</i>
              {views}
            </li>
            <li id="favorites"><i className="material-icons tiny">favorite</i>
              20
            </li>
            <li id="upvotes"><i className="material-icons tiny">thumb_up</i>
              {upvotes}
            </li>
            <li id="downvotes"><i className="material-icons tiny">thumb_down</i>
              {downvotes}
            </li>
          </ul>
          <a
            role="button"
            tabIndex="0"
            onClick={handleOpen}
            className={'btn-floating waves-effect waves-light green right'}
          >
            <i className="material-icons">edit</i>
          </a>
          <a
            role="button"
            tabIndex="0"
            onClick={handleOpenDelete}
            className={'btn-floating waves-effect waves-light red right'}
          >
            <i className="material-icons">delete</i>
          </a>
        </div>
      </div>
    </div>
  );
};

UserRecipeCard.propTypes = {
  recipe: PropTypes.shape({
    title: PropTypes.string,
    description: PropTypes.string,
    upvotes: PropTypes.number,
    downvotes: PropTypes.number,
  }).isRequired,
  handleOpenEdit: PropTypes.func.isRequired,
  handleOpenDelete: PropTypes.func.isRequired,
};

export default UserRecipeCard;
