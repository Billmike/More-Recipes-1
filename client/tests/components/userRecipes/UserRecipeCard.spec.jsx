import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import UserRecipeCard from '../../../components/userRecipes/UserRecipeCard';

describe('UserRecipeCard component', () => {
  it('should render component', () => {
    const props = {
      recipe: {
        title: 'Rice and beans',
        description: 'Tasty rice and beans',
        recipeImage: 'Recipe image URL',
        views: 54,
        upvotes: 32,
        downvotes: 2,
        favorites: 45,
      },
      handleOpenEdit: jest.fn(),
      handleOpenDelete: jest.fn(),
      currentProfileUserId: 3,
      authenticatedUserId: 1,
    };
    const wrapper = mount(
      <BrowserRouter><UserRecipeCard {...props} /></BrowserRouter>
    );

    expect(wrapper).toBeDefined();
    expect(wrapper.find('a#edit-recipe-button').exists()).toEqual(false);
    expect(wrapper.find('a#delete-recipe-button').exists()).toEqual(false);
    expect(wrapper).toMatchSnapshot();
  });
  it('should simulate edit and delete recipe click events', () => {
    const props = {
      recipe: {
        title: 'Rice and beans',
        description: 'Tasty rice and beans',
        recipeImage: 'Recipe image URL',
        views: 54,
        upvotes: 32,
        downvotes: 2,
        favorites: 45,
      },
      handleOpenEdit: jest.fn(),
      handleOpenDelete: jest.fn(),
      currentProfileUserId: 1,
      authenticatedUserId: 1,
    };
    const wrapper = mount(
      <BrowserRouter><UserRecipeCard {...props} /></BrowserRouter>
    );

    expect(wrapper).toBeDefined();
    expect(wrapper.find('a#edit-recipe-button').exists()).toEqual(true);
    expect(wrapper.find('a#delete-recipe-button').exists()).toEqual(true);
    wrapper.find('a#edit-recipe-button').simulate('click');
    expect(props.handleOpenEdit).toBeCalled();
    wrapper.find('a#delete-recipe-button').simulate('click');
    expect(props.handleOpenDelete).toBeCalled();
    expect(wrapper).toMatchSnapshot();
  });
});
