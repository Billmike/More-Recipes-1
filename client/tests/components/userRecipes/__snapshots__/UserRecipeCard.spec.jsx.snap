// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRecipeCard component should render component 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "blank",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <UserRecipeCard
      authenticatedUserId={1}
      currentProfileUserId={3}
      handleOpenDelete={[MockFunction]}
      handleOpenEdit={[MockFunction]}
      recipe={
        Object {
          "description": "Tasty rice and beans",
          "downvotes": 2,
          "favorites": 45,
          "recipeImage": "Recipe image URL",
          "title": "Rice and beans",
          "upvotes": 32,
          "views": 54,
        }
      }
    >
      <div
        className="col s12 m6 l3"
      >
        <div
          className="card large hoverable"
          id="recipes"
        >
          <Link
            replace={false}
            to="/recipes/undefined"
          >
            <a
              href="/recipes/undefined"
              onClick={[Function]}
            >
              <span
                className="card-title"
                id="recipe-title"
              >
                Rice and beans
              </span>
            </a>
          </Link>
          <div
            className="card-image"
          >
            <Link
              replace={false}
              to="/recipes/undefined"
            >
              <a
                href="/recipes/undefined"
                onClick={[Function]}
              >
                <RecipeImage
                  recipeImage="Recipe image URL"
                  title="Rice and beans"
                >
                  <img
                    alt="Rice and beans"
                    src="Recipe image URL"
                  />
                </RecipeImage>
              </a>
            </Link>
          </div>
          <div
            className="card-content"
          >
            <p>
              Tasty rice and beans
            </p>
          </div>
          <div
            className="card-action"
          >
            <ul
              className="center-align"
            >
              <li
                id="views"
              >
                <i
                  className="material-icons tiny"
                >
                  visibility
                </i>
                54
              </li>
              <li
                id="favorites"
              >
                <i
                  className="material-icons tiny"
                >
                  favorite
                </i>
                45
              </li>
              <li
                id="upvotes"
              >
                <i
                  className="material-icons tiny"
                >
                  thumb_up
                </i>
                32
              </li>
              <li
                id="downvotes"
              >
                <i
                  className="material-icons tiny"
                >
                  thumb_down
                </i>
                2
              </li>
            </ul>
            <ReactTooltip
              insecure={true}
              resizeHide={true}
              wrapper="div"
            >
              <div
                className="__react_component_tooltip place-top type-dark "
                data-id="tooltip"
              />
            </ReactTooltip>
          </div>
        </div>
      </div>
    </UserRecipeCard>
  </Router>
</BrowserRouter>
`;

exports[`UserRecipeCard component should simulate edit and delete recipe click events 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "blank",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <UserRecipeCard
      authenticatedUserId={1}
      currentProfileUserId={1}
      handleOpenDelete={
        [MockFunction] {
          "calls": Array [
            Array [
              undefined,
              "Are you sure you want to delete this recipe?",
              "Delete Recipe",
              "Delete",
            ],
          ],
        }
      }
      handleOpenEdit={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "description": "Tasty rice and beans",
                "downvotes": 2,
                "favorites": 45,
                "recipeImage": "Recipe image URL",
                "title": "Rice and beans",
                "upvotes": 32,
                "views": 54,
              },
            ],
          ],
        }
      }
      recipe={
        Object {
          "description": "Tasty rice and beans",
          "downvotes": 2,
          "favorites": 45,
          "recipeImage": "Recipe image URL",
          "title": "Rice and beans",
          "upvotes": 32,
          "views": 54,
        }
      }
    >
      <div
        className="col s12 m6 l3"
      >
        <div
          className="card large hoverable"
          id="recipes"
        >
          <Link
            replace={false}
            to="/recipes/undefined"
          >
            <a
              href="/recipes/undefined"
              onClick={[Function]}
            >
              <span
                className="card-title"
                id="recipe-title"
              >
                Rice and beans
              </span>
            </a>
          </Link>
          <div
            className="card-image"
          >
            <Link
              replace={false}
              to="/recipes/undefined"
            >
              <a
                href="/recipes/undefined"
                onClick={[Function]}
              >
                <RecipeImage
                  recipeImage="Recipe image URL"
                  title="Rice and beans"
                >
                  <img
                    alt="Rice and beans"
                    src="Recipe image URL"
                  />
                </RecipeImage>
              </a>
            </Link>
          </div>
          <div
            className="card-content"
          >
            <p>
              Tasty rice and beans
            </p>
          </div>
          <div
            className="card-action"
          >
            <ul
              className="center-align"
            >
              <li
                id="views"
              >
                <i
                  className="material-icons tiny"
                >
                  visibility
                </i>
                54
              </li>
              <li
                id="favorites"
              >
                <i
                  className="material-icons tiny"
                >
                  favorite
                </i>
                45
              </li>
              <li
                id="upvotes"
              >
                <i
                  className="material-icons tiny"
                >
                  thumb_up
                </i>
                32
              </li>
              <li
                id="downvotes"
              >
                <i
                  className="material-icons tiny"
                >
                  thumb_down
                </i>
                2
              </li>
            </ul>
            <a
              className="right"
              id="edit-recipe-button"
              onClick={[Function]}
              role="button"
              tabIndex="0"
            >
              <i
                className="material-icons icon-green"
                data-tip="Edit Recipe"
              >
                edit
              </i>
            </a>
            <a
              className="right"
              id="delete-recipe-button"
              onClick={[Function]}
              role="button"
              tabIndex="0"
            >
              <i
                className="material-icons icon-red"
                data-tip="Delete Recipe"
              >
                delete
              </i>
            </a>
            <ReactTooltip
              insecure={true}
              resizeHide={true}
              wrapper="div"
            >
              <div
                className="__react_component_tooltip place-top type-dark "
                data-id="tooltip"
              />
            </ReactTooltip>
          </div>
        </div>
      </div>
    </UserRecipeCard>
  </Router>
</BrowserRouter>
`;
